index.js

```diff
@@ -46,230 +48,230 @@ function resolveInputDirectory(inputNodes) {
+    results = self.processResults(results, relativePath);
+    if(results.errored && self.testFailingFiles) {
+    results.output = self.testGenerator(relativePath, results);
+    } else if(!results.errored && self.testPassingFiles) {
+      results.output = self.testGenerator(relativePath);
+    }
+    return results;
+  }).catch(function(err) {
+    console.error(chalk.red('======= Something went wrong running stylelint ======='));
+    if(err.code === 78){
+      if(err.message.indexOf('No configuration provided') > -1){
+        console.error(chalk.red("No stylelint configuration found please create a .stylelintrc file in the route directory"));
+      } else {
+        console.error(chalk.red(err.message));
       }
-      if(this.log)
-       this.consoleLogger(results, relativePath);
-     }
-     return results;
-  } else {
-    return {};
+    } else {
+      console.error(err.stack);
+    }
+    console.error(err.stack);
+  });
   }
-};
-
-/**
-  * @method processResults
-  *
-  *  Reformats default results object
-  *  {
-  *   errored: boolean if file errored or not,
-  *   output: String contains test if generate test is true,
-  *   log: String default logging string,
-  *   source: String relitivePath,
-  *   deprecations: Array of errors,
-  *   invalidOptionWarnings: Array,
-  *   warnings: Array of errors,
-  *   ignored: Array ignored files,
-  *   _postcssResult: Object for postcss
-  *  }
-  */
-StyleLinter.prototype.processResults = function(results, relativePath) {
- var resultsInner = results.results[0];
- resultsInner.errored = results.errored;
- resultsInner.source = relativePath;
- delete results.results;
- results.log = results.output;
- Object.assign(results, resultsInner);
- results.source = relativePath;
- results.output = '';
- return results;
-};
 
-/**
-  * @method testGenerator
-  *
-  *  Alias of escapeString for hooks
-  */
-StyleLinter.prototype.escapeErrorString = escapeString;
+  /**
+    * @method postProcess
+    * This method is called after, the file has been linted:
+    *  - Logs to console
+    *  - Generate tests
+    * @override
+    */
+  postProcess(results, relativePath) {
+    if(results) {
+      if(results.errored){
+        if(this.onError) {
+        this.onError(results);
+        }
+        if(this.log)
+        this.consoleLogger(results, relativePath);
+      }
+      return results;
+    } else {
+      return {};
+    }
+  }
 
+  /**
+    * @method processResults
+    *
+    *  Reformats default results object
+    *  {
+    *   errored: boolean if file errored or not,
+    *   output: String contains test if generate test is true,
+    *   log: String default logging string,
+    *   source: String relitivePath,
+    *   deprecations: Array of errors,
+    *   invalidOptionWarnings: Array,
+    *   warnings: Array of errors,
+    *   ignored: Array ignored files,
+    *   _postcssResult: Object for postcss
+    *  }
+    */
+  processResults(results, relativePath) {
```

Comment:
need to add some spacing to the contents of this function

Reply:
done